### PR TITLE
Floating Visualizer Fixes

### LINKED ISSUE #13566: Floating Animation resets upon re pvs playing the animation when gravity is on
repro
have grav turn off
have grav go back on
leave and reenter pvs range of someone affected
everytime you do the last loop of the animation will play

### LINKED ISSUE #13572: Guidebook monkey experiences 0g and thus floats out of its spot
help him
image

### PR DESCRIPTION
**Pull request by AJCM-git**
## About the PR
Fixes #13572 
Fixes #13566

Added a nullspace check to IsWeightless

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- fix: Neither monkeys on the guidebook nor people outside your view float

### CODE REVIEW

**Comment by ElectroJr** [comment]
Entities in null space just shouldn't be floating in the first place.

**Comment by AJCM-git** [comment]
> Entities in null space just shouldn't be floating in the first place.

Okay that was the core issue, fixed (someone merge for me cuz im gonna snooze)

File: Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs [hunk 1]
```diff
@@ -42,6 +43,9 @@ private void OnGravityChanged(ref GravityChangedEvent args)
     {
         foreach (var (floating, transform) in EntityQuery<FloatingVisualsComponent, TransformComponent>(true))
         {
+            if (transform.MapID == MapId.Nullspace)
+                return;
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedFloatingVisualizerSystem.cs:47]:
```suggestion
            if (transform.MapID == MapId.Nullspace)
                continue;
```

File: Content.Shared/Gravity/SharedGravitySystem.cs [hunk 2]
```diff
@@ -31,8 +31,9 @@ public bool IsWeightless(EntityUid uid, PhysicsComponent? body = null, Transform
                 return true;
 
             // If grid / map has gravity
-            if (TryComp<GravityComponent>(xform.GridUid, out var gravity) && gravity.Enabled ||
-                 TryComp<GravityComponent>(xform.MapUid, out var mapGravity) && mapGravity.Enabled)
+            if (xform.MapID == MapId.Nullspace ||
```
Comment (metalgearsloth) [inline Content.Shared/Gravity/SharedGravitySystem.cs:34]: This seems kinda suss when floating visualizer shouldn't be applying in nullspace?

**Review by metalgearsloth — COMMENTED** [review]
Should you sub to entparentchangedmessage for floatingvisualscomponent as well?

**Pull request merged**